pkg/agent/loadbalancer/loadbalancer.go

```diff
@@ -117,7 +95,7 @@ func New(ctx context.Context, dataDir, serviceName, serverURL string, lbServerPo
 	}
 	lb.proxy.AddRoute(serviceName, &tcpproxy.DialProxy{
 		Addr:        serviceName,
-		DialContext: lb.dialContext,
+		DialContext: lb.servers.dialContext,
 		OnDialError: onDialError,
 	})
```

Comment:
Trying to understand the logic of when the Full `DialContext` will be called. We have a `DialContext` for all servers which uses the wrapped conneciton of each servers individual `DialContext` method + records the result in the `RecordSuccess/Failure`, potential changing the state of the server (and thus all servers in the list).  When does this Full call happen?

Reply:
tcpproxy listens for connections, and calls the DialContext function here when one is accepted, in order to make a connection to a backend server. The lb.servers.DialContext function loops through the server list and calls DialContext on each of them, returning the first one that succeeds. tcpproxy then passes data back and forth between the two connections.